dispersion_protocol/dispersion.xml

```diff
@@ -0,0 +1,184 @@
+        <field type="uint32_t" name="hardware_version">Version of the payload hardware, encoded as: (Dev & 0xff) << 24</field>
+        <field type="uint64_t" name="uid">Integer to uniquely identify this hardware. (0 if unknown)</field>
+        <field type="uint16_t" name="cap_flags" enum="DISPERSION_DEVICE_CAP_FLAGS", display="bitmask">Bitmap of dispersion device capability flags.</field>
+        <field type="uint16_t" name="custom_cap_flags" display="bitmask">Bitmap for use for dispersion-specific capability flags. flags.</field>
+        <field type="char[64]" name="logger_dir_ftp_url">File transfer protocol (FTP) url-like string pointing to the directory of log files</field>
+        <field type="float" name="dispersion_rate_min" units="liters/min or kg/min">Minimum dispersion rate this device can support</field>
+        <field type="float" name="dispersion_rate_max" units="liters/min or kg/min">Maximum dispersion rate this device can support</field>
+        <field type="uint32_t" name="pressure_min" units="Pa">If this is a spray dispersion system, minimum pressure the hardware supports independent of nozzle configuration</field>
+        <field type="uint32_t" name="pressure_max" units="Pa">If this is a spray dispersion system, maximum pressure the hardware supports independent of nozzle configuration</field>
+        <field type="float" name="max_capacity" units="liters or kg">Maximum fill capacity of this device in Liters for sprayers and Kg for spreaders</field>
+    </message>
+    <message id="361" name="DISPERSION_DEVICE_STATUS">
+      <description>Message reporting the status of a dispersion device. This message should be published at a low regular rate (e.g. 5 Hz) but also during key events such as a mavlink command, flag change, or rapid pressure change.</description>
+      <field type="uint64_t" name="timestamp_ns" units="ns">Unix time stamp in nanoseconds</field>
+      <field type="uint32_t" name="dispersion_type", enum="MAV_DISPERSION_TYPE">Dispersion type. Defines units for device capacity and dispersion rate</field>
+      <field type="uint16_t" name="flags" enum="DISPERSION_DEVICE_STATUS_FLAGS" display="bitmask">Current flags set by the device</field>
+      <field type="uint32_t" name="failure_flags" enum="DISPERSION_DEVICE_ERRORS" display="bitmask">Bitmap for failure flags (0 for no failure). Any failure indicates the system has stopped</field>
+      <field type="uint32_t" name="warning_flags" enum="DISPERSION_DEVICE_WARNINGS" display="bitmask">Bitmap for warning flags (0 for no warning)</field>
+      <field type="uint8_t" name="custom_codes">Leave some flexibility for device manufacturers to pass more information through. This gives up to 255 codes to represent internal system state.</field>
+      <field type="float" name="dispersion_rate" units="liters/min or kg/min">Current dispersion rate of the device. Negative during refill.</field>
```

Comment:
Not sure if double units like these will work with our parsing/tooling. Probably needs to be unitless and then just have units in the comment.

Reply:
They won't work - the validation will fail anything except the supported set.

I would remove this units, and add the information in the description (with a comment that the units depends on dispersion_type).